### PR TITLE
refactoring: privacy indicator

### PR DESCRIPTION
**Pull request by aleksandr-cliqz**
@chrmod @shahenAnt @palewar 
guys, could you review plz, ~~meanwhile I am adding some tests (as I understand it's not very easy task :)~~ 
~~Also, could anybody help me how to test `addStrikeThroughToChart` in the interface~~

~~ALSO! not sure about ghostery, can easily miss some files~~

UPD: done 

Fix #

## Checklist:

- [ ] My squashed commit messages follows the [7 golden rules](https://chris.beams.io/posts/git-commit/)
- [x] I updated or created necessary unit tests
- [x] I've tested my changes in Dark Mode, iPad, iOS <= 12, Landscape and in German

### CODE REVIEW

**Comment by aleksandr-cliqz** [comment]
@chrmod @palewar @shahenAnt 
guys, i checked it with 1) cliqz.com, 2) focus.de 3) disabled indicator 4) iphone 8, 5) iphone 11, 5) iPad, 6) landscape orientation 
Have no idea how to check it with dark mode and IOS <= 12 :)

**Comment by aleksandr-cliqz** [comment]
@chrmod @palewar @shahenAnt 
guys, could you take a look please ?

File: UserAgent/Views/Privacy Indicator/PrivacyIndicatorCanvasView.swift [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    private var cachedStats: [WTMCategory: Int] = [:]
+    private var backgroundTrackLayer: CAShapeLayer?
+    private var greenLayer: CAShapeLayer?
+    private var strikeThroughLayer: CAShapeLayer?
+}
+
+extension PrivacyIndicatorCanvasView {
+    func render(with stats: TPPageStats, status: BlockerStatus) {
+        DispatchQueue.main.async { [weak self] in
+            if self == nil { return }
```
Comment (shahenAnt) [inline UserAgent/Views/Privacy Indicator/PrivacyIndicatorCanvasView.swift]: As you don't need to force unwrap self in this scope so you don't need to check "self" is nil or not. The dispatch scope can be:
{ [weak self] in
     self?.cachedStats = WTMCategory.statsDict(from: stats)
     self?.render(status: status)
}

Comment (shahenAnt) [reply UserAgent/Views/Privacy Indicator/PrivacyIndicatorCanvasView.swift]: @chrmod This is not blocking comment you can merge it.

Comment (aleksandr-cliqz) [reply UserAgent/Views/Privacy Indicator/PrivacyIndicatorCanvasView.swift]: @shahenAnt make sense, thx

**Pull request merged**